packages/store/ts/config/v2/input.ts

```diff
@@ -18,11 +18,17 @@ export type TableDeployInput = Partial<TableDeploy>;
 
 export type TableInput = {
   /**
-   * Human-readable table label. Used as config keys, table library names, and filenames.
+   * Human-readable label for this table. Used as config keys, library names, and filenames.
    * Labels are not length constrained like resource names, but special characters should be avoided to be compatible with the filesystem, Solidity compiler, etc.
    */
   readonly label: string;
   /**
+   * Human-readable label for this table's namespace. Used for namespace config keys and directory names.
+   * Defaults to the nearest namespace in the config or root namespace if not set.
+   */
+  readonly namespaceLabel?: string;
```

Comment:
probably too big of a change now and not worth the effort but wonder if defining namespace as its own type would make sense e.g:

type NamespaceInput = {
  id: string;
  label?: string;
}

type Table Input = {
  namespace?: NamespaceInput;
  // ...
}

Reply:
this would be a breaking change unfortunately